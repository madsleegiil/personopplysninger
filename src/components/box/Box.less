.box {
  &__container {
    display: flex;
    flex-flow: row;
    background: white;
    margin: 0.25rem 0;
    padding: 1.5rem 2rem;

    @media @mobile {
      flex-direction: column;
    }

    @media @tablet {
      padding: 1.5rem 2rem;
    }

    @media @desktop {
      padding: 1.75rem 3rem;
    }
  }

  &__content-container {
    width: 100%;
  }

  &__header {
    display: flex;
    flex-flow: column;
  }

  &__header > * {
    margin: 0.4rem 0;
  }
  &__ingress {
    max-width: 100%;
  }

  &__icon {
    width: 4rem;
    height: 4rem;
    @media @mobile {
      visibility: hidden;
    }
  }

  &__linje-smal {
    border: 1px solid black;
    width: 15%;
  }

  &__linje-bred {
    border: 0.5px solid black;
    width: 90%;
    opacity: 0.5;
    margin: 1.4rem auto;
  }

  &__content {
    ul {
      display: flex;
      flex-direction: row;
      flex-wrap: wrap;
      list-style-type: none;
      padding-left: 0;
    }

    li {
      flex-grow: 1;
      box-sizing: border-box;
      padding: 0rem 2rem 1rem 0rem;

      .title {
        display: block;
        white-space: nowrap;
        text-overflow: ellipsis;
        overflow: hidden;
      }
    }

    ul.list-column-1 li {
      width: 100%;
    }

    ul.list-column-2 li {
      width: 50%;
      @media @mobile {
        width: 100%;
      }
    }

    ul.list-column-2.address-columns {
      li:nth-child(1) {
        order: 1;
      }

      li:nth-child(2) {
        order: 3;
      }

      li:nth-child(3) {
        order: 5;
      }

      li:nth-child(4) {
        order: 2;
      }

      li:nth-child(5) {
        order: 4;
      }
    }

    ul.list-column-3 li {
      width: 33%;
      @media @mobile {
        width: 100%;
      }
    }

    ul.list-column-4 li {
      width: 25%;
      @media @mobile {
        width: 100%;
      }
    }
  }
}

//midlertidig inntil avklaring
.information-circle {
  margin: 0 0.5rem;
  border: none;
  background: none;
}

.infobox {
  padding: 2em;
  background-color: #ffe5c2;
  display: inline-block;
  border-radius: 4px;
  margin-bottom: 2em;

  hr {
    display: block;
    height: 1px;
    border: 0;
    border-top: 1px solid #ff9100;
    padding: 0;
  }

  h2 {
    text-align: center;
    margin-bottom: 4rem;
  }

  .info-content {
    text-align: center;
    margin-bottom: 2rem;
  }
}
