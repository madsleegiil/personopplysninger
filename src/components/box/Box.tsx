import React, { useState, useEffect } from "react";
import { Ingress, Systemtittel } from "nav-frontend-typografi";
import Ekspanderbartpanel from "nav-frontend-ekspanderbartpanel";
import { FormattedMessage, injectIntl, InjectedIntlProps } from "react-intl";
import Panel from "nav-frontend-paneler";
import Icon from "../icon/Icon";

interface Props {
  id: string;
  tittel: string;
  beskrivelse?: string;
  icon?: string;
  children: JSX.Element | JSX.Element[];
}

const Box = (props: Props & InjectedIntlProps) => {
  const { tittel, beskrivelse, icon, children, id, intl } = props;
  const [erMobil, setErMobil] = useState(window.innerWidth <= 420);

  const desktopVersjon = (
    <Panel className="box">
      <div className="icon__container" id={id}>
        {icon && <Icon backgroundImage={icon} />}
      </div>
      <div className="box__content-container">
        <div className="box__header" id={id}>
          {tittel && (
            <Systemtittel>
              <FormattedMessage id={tittel} />
            </Systemtittel>
          )}
          {beskrivelse && (
            <div className="box__ingress">
              <Ingress>
                <FormattedMessage id={beskrivelse} />
              </Ingress>
            </div>
          )}
        </div>
        <div className="box__content">{children}</div>
      </div>
    </Panel>
  );

  const mobilVersjon = (
    <div className="box" id={id}>
      <Ekspanderbartpanel
        tittel={intl.formatMessage({ id: tittel })}
        tittelProps="systemtittel"
      >
        <div className="box">
          {beskrivelse && (
            <div className="box__ingress">
              <Ingress>
                <FormattedMessage id={beskrivelse} />
              </Ingress>
              <hr className="box__linje-smal" />
            </div>
          )}
          <div className="box__content">{children}</div>
        </div>
      </Ekspanderbartpanel>
    </div>
  );

  useEffect(() => {
    window.addEventListener("resize", () =>
      setErMobil(window.innerWidth <= 420)
    );
    return () => {
      window.removeEventListener("resize", () =>
        setErMobil(window.innerWidth <= 420)
      );
    };
  });

  return <>{erMobil ? mobilVersjon : desktopVersjon}</>;
};

export default injectIntl(Box);
