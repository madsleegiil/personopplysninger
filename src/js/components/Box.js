import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Box extends Component {
  render() {
    return (
      <div className="BoxContainer">
        <div className="BoxWithHeader">
          <h1>{this.props.header}</h1>
          <div className="icon-box-wrapper">
            {this.props.icon ? <img src={this.props.icon} alt="" className="box-icon" /> : null}
            <div className="Box">
              {this.props.children}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Box.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  header: PropTypes.string,
  icon: PropTypes.string,
};

Box.defaultProps = {
  children: [],
  header: '',
  icon: '',
};

export default Box;
