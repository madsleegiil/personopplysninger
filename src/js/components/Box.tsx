import React, { useState, useEffect } from "react";
import { Ingress, Systemtittel } from "nav-frontend-typografi";
import Ekspanderbartpanel from "nav-frontend-ekspanderbartpanel";
import Panel from "nav-frontend-paneler";
import infoContent from "../static/infoContent";

const Box = (props: any) => {
  const { header, icon, infoType, children, id } = props;
  const [erMobil, setErMobil] = useState(window.innerWidth <= 420);

  const desktopVersjon = (
    <Panel className="box">
      <div className="box__header" id={id}>
        {icon ? <img src={icon} alt="" className="box__icon" /> : null}
        {header ? <Systemtittel>{header}</Systemtittel> : null}
        {infoType ? (
          <>
            <hr className="box__linje-smal" />
            <Ingress>{infoContent[infoType].content}</Ingress>
            <hr className="box__linje-bred" />
          </>
        ) : null}
      </div>
      <div className="box__content">{children}</div>
    </Panel>
  );

  const mobilVersjon = (
    <div className="box" id={id}>
      <Ekspanderbartpanel tittel={header} tittelProps="systemtittel">
        <div className="box">
          {infoType ? (
            <>
              <Ingress>{infoContent[infoType].content}</Ingress>
              <hr className="box__linje-smal" />
            </>
          ) : null}
          <div className="box__content">{children}</div>
        </div>
      </Ekspanderbartpanel>
    </div>
  );

  useEffect(() => {
    window.addEventListener("resize", () =>
      setErMobil(window.innerWidth <= 420)
    );
    return () => {
      window.removeEventListener("resize", () =>
        setErMobil(window.innerWidth <= 420)
      );
    };
  });

  return <>{erMobil ? mobilVersjon : desktopVersjon}</>;
};

export default Box;
