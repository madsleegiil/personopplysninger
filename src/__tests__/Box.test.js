import * as React from 'react';
import Box from 'js/components/Box';
import wrapIntl from 'js/IntlTestHelper';
const ReactTestRenderer = require('react-test-renderer');

test('render Header with content', () => {
  const component = ReactTestRenderer.create(wrapIntl(<Box header="Arbeidsforhold">
    <h3>Header</h3>
    <div>content</div>
  </Box>));
  expect(component.toJSON()).toMatchSnapshot();
});
