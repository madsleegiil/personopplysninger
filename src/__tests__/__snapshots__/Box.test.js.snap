// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header with content 1`] = `
<div
  className="BoxContainer box-bottom-margin"
>
  <div
    className="BoxWithHeader"
  >
    <h1>
      Arbeidsforhold
    </h1>
    <div
      className="icon-box-wrapper"
    >
      <div
        className="box-icon"
      >
        
      </div>
      <div
        className="Box"
      >
        <h3>
          Header
        </h3>
        <div>
          content
        </div>
      </div>
    </div>
  </div>
</div>
`;
